device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsSessionHandler.java

```diff
@@ -80,13 +80,35 @@ public void setSession(Session session)
         this.twinReceiverLinkOpened = false;
         this.methodsSenderLinkOpened = false;
         this.methodsReceiverLinkOpened = false;
+        this.sessionOpenedRemotely = false;
+        this.sessionHandlerClosedBeforeRemoteSessionOpened = false;
     }
 
     public void closeSession()
     {
         if (this.session != null)
         {
-            this.session.close();
+            if (this.sessionOpenedRemotely)
```

Comment:
Is this ever being set somewhere or is this for the future?

Reply:
Good catch, somehow this didn't get committed to this PR. I've added it being set to true on the onSessionRemoteOpen event now